dictdiffer/unify.py

```diff
@@ -37,7 +37,8 @@ def unify(self, first_patches, second_patches, conflicts):
                 if conflict.take_patch() != patch:
                     continue
 
-            self.unified_patches.append(patch)
+            if patch not in self.unified_patches:
```

Comment:
This might be quite slow if you have a long list of `unified_patches`.

Reply:
Do you have any suggestions perhaps?

Reply:
I would even propose a change to this method:

```
unified_patches = set()
self._build_index(conflicts)

sorted_patches = sorted(first_patches + second_patches, key=get_path)

for patch in sorted_patches:
    conflict = self._index.get(nested_hash(patch))

    # Apply only the patches that were taken as part of conflict
    # resolution.
    if conflict:
        if conflict.take_patch() != patch:
            continue

    if patch not in unified_patches:
        unified_patches.add(patch)
        yield patch
```

Reply:
also `_build_index` method could become static and simply return the conflict mapping.

Reply:
@jirikuncar I think maybe before this [commit](https://github.com/inveniosoftware/dictdiffer/commit/e637f35a2a5e7fb53f6a554cd07c51e4baa0a84c) this bug was not here?

Reply:
@iulianav maybe instead of sorting the lists should be merged.

```python
>>> a = [1, 4, 3, 2]
>>> b = [1, 5, 3]
>>> merge_keys(a, b)
[1, 4, 5, 3, 2]
```